components/director/internal/statusupdate/handler.go

```diff
@@ -0,0 +1,120 @@
+package statusupdate
+
+import (
+	"context"
+	"encoding/json"
+	"net/http"
+
+	"github.com/pkg/errors"
+
+	"github.com/kyma-incubator/compass/components/director/pkg/persistence"
+
+	log "github.com/sirupsen/logrus"
+
+	"github.com/kyma-incubator/compass/components/director/internal/consumer"
+)
+
+type update struct {
+	transact persistence.Transactioner
+	repo     StatusUpdateRepository
+	logger   *log.Logger
+}
+
+type Table string
+
+//go:generate mockery -name=StatusUpdateRepository -output=automock -outpkg=automock -case=underscore
+type StatusUpdateRepository interface {
+	UpdateStatus(ctx context.Context, id, table string) error
+	IsConnected(ctx context.Context, id, table string) (bool, error)
+}
+
+const (
+	applicationsTable Table = "applications"
```

Comment:
handler should not be aware of table names, it will be better to have separate methods for app and runtimes (IsApplicationConnected) or introduce some enum.